dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/TaskCreator.scala

```diff
@@ -93,49 +101,71 @@ class TaskCreator(context: EngineContext) {
       .distinct
 
     val forCalls = outArgsAndCalls.flatMap { case (result, outArg, path, callDepth) =>
-      val outCall = outArg.collect { case n: Call => n }
-
-      val methodReturns = outCall.toList
-        .flatMap(x => NoResolve.getCalledMethods(x).methodReturn.map(y => (x, y)))
-        .iterator
-
-      methodReturns.flatMap { case (call, methodReturn) =>
-        val method           = methodReturn.method
-        val returnStatements = methodReturn._reachingDefIn.toList.collect { case r: Return => r }
-        if (method.isExternal || method.start.isStub.nonEmpty) {
-          val newPath = path
-          (call.receiver.l ++ call.argument.l).map { arg =>
-            val taskStack = result.taskStack :+ TaskFingerprint(arg, result.callSiteStack, callDepth)
-            ReachableByTask(taskStack, newPath)
-          }
-        } else {
-          returnStatements.map { returnStatement =>
-            val newPath = Vector(PathElement(methodReturn, result.callSiteStack)) ++ path
-            val taskStack =
-              result.taskStack :+ TaskFingerprint(returnStatement, call :: result.callSiteStack, callDepth + 1)
-            ReachableByTask(taskStack, newPath)
+      outArg.toList.flatMap {
+        case call: Call =>
```

Comment:
This first part looks to me as if the `.collect` was just turned into an equivalent `.flatMap`. Can you explain if that is the only change up to line 130?

Reply:
Yes, only formatting and handling of `case _ => Vector.empty` for nonCall nodes